src/utils/time.js

```diff
@@ -31,3 +31,95 @@ export function isDateNotValid(d) {
     return true;
   }
 }
+
+export function isZeroDateTime(date) {
+  return !date || (date.startsWith && date.startsWith('0001-01-01T'));
+}
+
+export function calculateDuration(duration, type = EXTENDED_TIME_PERIODS.MILISECONDS) {
+  let divider;
+  switch (type) {
+    case EXTENDED_TIME_PERIODS.MILISECONDS:
+      divider = 1000 * 60;
+      break;
+    case EXTENDED_TIME_PERIODS.MINUTES:
+      divider = 1;
+      break;
+    case EXTENDED_TIME_PERIODS.SECONDS:
+    default:
+      divider = 60;
+  }
+  let diff = duration;
+  const days = Math.floor(diff / 24 / divider / 60).toString();
+  diff -= days * 24 * 60 * divider;
+  let hours = Math.floor(diff / 60 / divider);
+  diff -= hours * 60 * divider;
+  let minutes = Math.floor(diff / divider);
+  diff -= minutes * divider;
+  let seconds = 0;
+  if (type !== EXTENDED_TIME_PERIODS.MINUTES && type !== EXTENDED_TIME_PERIODS.DAYS) {
+    seconds = type === EXTENDED_TIME_PERIODS.SECONDS ? diff : Math.floor(diff / 1000);
+  }
+  hours = (hours <= 9 ? '0' : '') + hours;
+  minutes = (minutes <= 9 ? '0' : '') + minutes;
+  seconds = (seconds <= 9 ? '0' : '') + seconds;
+  return {
+    days,
+    hours,
+    minutes,
+    seconds
+  };
+}
+
+export function getDurationString(duration, type) {
+  const { days, hours, minutes } = calculateDuration(duration, type);
+  const daysNumber = parseInt(days, 10);
+  let timeToRender = `${daysNumber.toString()}d ${hours}h ${minutes}m`;
+  if (daysNumber === 0) {
+    timeToRender =
+      `${hours > 0 && hours !== '0' ? `${hours}h ` : ''}` +
+      `${minutes}m`;
+  }
+  return timeToRender;
+}
+
+function getSlaEffectiveStartTime(timerValue) {
+  return moment(timerValue.startDate).add(timerValue.accumulatedPause, EXTENDED_TIME_PERIODS.SECONDS);
+}
+
+function getSlaEffectiveStopDate(timerValue) {
+  return timerValue.runStatus === TIMER_STATUS.ended ? timerValue.endDate : timerValue.lastPauseDate;
+}
+function getSlaTimerTimeRan(timerValue) {
+  const startDate = getSlaEffectiveStartTime(timerValue);
+  const stopDate = getSlaEffectiveStopDate(timerValue);
+
+  return moment(stopDate).diff(startDate, EXTENDED_TIME_PERIODS.SECONDS);
+}
+
+function getSlaTimeLeft(timerValue) {
+  const effectiveStopDate = getSlaEffectiveStopDate(timerValue);
+
+  return moment(timerValue.dueDate).diff(effectiveStopDate, EXTENDED_TIME_PERIODS.SECONDS);
```

Comment:
```suggestion
  return moment(timerValue.dueDate).diff(getSlaEffectiveStopDate(timerValue), EXTENDED_TIME_PERIODS.SECONDS);
```

Reply:
It is like this in the web-client. I think both of the way to right code are ok (no preferred way).